src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -622,11 +632,47 @@ private void CreateTombstone(DocumentsOperationContext context, Table collection
             table.Insert(tbv);
         }
 
-        public bool DeleteCollection(DocumentsOperationContext context, string name, List<long> deletedList, long untilEtag)
+        public void DeleteCollection(DocumentsOperationContext context, string name, BlittableJsonTextWriter writer)
         {
+            _versioningStorage.DeleteCollection(context, name);
+
             name = "@" + name; //todo: avoid this allocation
-            var table = new Table(_docsSchema, name, context.Transaction.InnerTransaction);
-            return table.DeleteAll(_docsSchema.FixedSizeIndexes["CollectionEtags"], deletedList, untilEtag);
+
+            var deletedList = new List<long>();
+            long totalDocsDeletes = 0;
+            long maxEtag = -1;
+            while (true)
+            {
+                bool isAllDeleted;
+                using (context.OpenWriteTransaction())
+                {
+                    if (maxEtag == -1)
+                        maxEtag = ReadLastEtag(context.Transaction.InnerTransaction);
+
+                    var table = new Table(_docsSchema, name, context.Transaction.InnerTransaction);
+
+                    isAllDeleted = table.DeleteAll(_docsSchema.FixedSizeIndexes["CollectionEtags"], deletedList, maxEtag);
+                    context.Transaction.Commit();
+                }
+                context.Write(writer, new DynamicJsonValue
+                {
+                    ["BatchSize"] = deletedList.Count
+                });
+                writer.WriteComma();
+                writer.WriteNewLine();
+                writer.Flush();
+
+                totalDocsDeletes += deletedList.Count;
+
+                if (isAllDeleted)
+                    break;
+
+                deletedList.Clear();
+            }
+            context.Write(writer, new DynamicJsonValue
```

Comment:
I agree that it is not the best solution, but it is better than being on the controller, and calling DeleteCollection for versioning from there as well. 

DeleteCollection is special in the case that we split this action to a few transactions, and we want to write the status to a stream.